### PR TITLE
Add Notepad++ External Editor for Windows

### PR DESCRIPTION
**Pull request by ergaziz**

### CODE REVIEW

**Comment by niik** [comment]
Hey @ergaziz, seems like you got a few conflicts here after I merged #3962. Mind taking a stab at resolving those? Please let us know if you'd like any assistance.

**Comment by ergaziz** [comment]
hey @niik ,
No problem, just tried to solve on web interface. Can you check out if it
is resolved or not?

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -258,6 +283,13 @@ function extractApplicationInformation(
     return { displayName, publisher, installLocation }
   }
 
+  if (editor === ExternalEditor.NotepadPlusPlus) {
+    const displayName = getKeyOrEmpty(keys, 'DisplayName')
+    const publisher = getKeyOrEmpty(keys, 'Publisher')
+    const installLocation = getKeyOrEmpty(keys, 'DisplayIcon')
```
[the remark this answers is not in the document]
Comment (ergaziz) [reply app/src/lib/editors/win32.ts]: @niik , yes it doesn't seems good to me either but Notepad++ does not put `InstallLocation` key unfortunately.
Another option is to use `UninstallString` which gives `C:\Program Files (x86)\Notepad++\uninstall.exe` and find our executable relative to that. 
Do you think we should go for that?

We check for executable in Line 324 : 
```Typescript
  const path = getExecutableShim(editor, installLocation)
  const exists = await pathExists(path)
  if (!exists) {
    log.debug(`Command line interface for ${editor} not found at '${path}'`)
    return null
  }
``` 
Is it a sufficient check ?

**Comment by ergaziz** [comment]
We need to open notepad++ with `-openSession` argument in order to open repo in workspace instead of individual files.
```bash
C:\Program Files (x86)\Notepad++>"notepad++.exe" -openSession C:\myrepo 
```
Working on it.

**Comment by ergaziz** [comment]
We are blocked here due to [notepad++](https://github.com/notepad-plus-plus/notepad-plus-plus) unfortunately. Turns out that we were never able to open notepad++ in "Folder as Workspace" mode from command line. It can be done via File-> "Open Folder as Workspace" only.
npp_folderasws

Opened issue to notepad++ [here](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/4253)

For now, there seems like nothing we can do and I think we can cancel this issue. @niik , @iAmWillShepherd what do you think?

**Comment by niik** [comment]
> For now, there seems like nothing we can do and I think we can cancel this issue. @niik , @iAmWillShepherd what do you think?

I'm afraid I know nothing about Notepad++. What's the behavior with this current solution? Does it just open in a new window? I think you'll know best whether that's better than not supporting at all.

**Comment by ergaziz** [comment]
I didn't know much either, just some of my friends are using it, not me... :smiley: 
Current solution tries to open each file in the repo as individual files. I think it doesn't bring much functionality, it causes such things instead : 
desktop_npp

**Comment by j-f1** [comment]
@ergaziz I’d recommend closing this PR for now, then reopening it and picking up where you left off once they add support.

**Comment by Seb33300** [comment]
Hello, the feature has been added to the latest Notepad++ 7.7.1 release:
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/57a49c3ba5816f74827784830dbe7c5e494f397c

**Comment by Seb33300** [comment]
Or maybe this has been added only to Notpad++ 7.8 regarding the changelog:

> 4.  Add the command line argument "-openFoldersAsWorkspace" to open folders in "folder as workspace" panel.
    Example: notepad++ -openFoldersAsWorkspace c:\src\myProj01 c:\src\myProj02

**Pull request closed without merging**